upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -700,6 +716,39 @@ func (c *ApplyClusterCmd) Run() error {
 
 	tf.AddTo(l.TemplateFunctions)
 
+	if cluster.Spec.Assets != nil && cluster.Spec.Assets.ContainerRegistry != nil {
+		kubernetesComponents := []string{"kube-proxy", "kube-apiserver", "kube-controller-manager", "kube-scheduler"}
```

Comment:
Anything that calls `Image` above should do this automatically.  That is called by proxy, kcm & scheduler.

kube-apiserver is still using the models/config/kube-apiserver (!)